apps/managedidentity/managedidentity.go

```diff
@@ -173,17 +280,104 @@ func createIMDSAuthRequest(ctx context.Context, id ID, resource string, claims s
 	return req, nil
 }
 
-func (client Client) getTokenForRequest(req *http.Request) (accesstokens.TokenResponse, error) {
+func createAzureArcAuthRequest(ctx context.Context, id ID, resource string, claims string) (*http.Request, error) {
+	identityEndpoint, _ := getAzureArcEnvironmentVariables()
+	var msiEndpoint *url.URL
+
+	if identityEndpoint == "" {
+		identityEndpoint = azureArcEndpoint
+	}
+
+	if _, ok := id.(systemAssignedValue); !ok {
+		return nil, errors.New("Azure Arc doesn't support specifying a user-assigned managed identity at runtime")
+	}
+
+	identityEndpoint = strings.Replace(identityEndpoint, "localhost", "127.0.0.1", -1)
+	msiEndpoint, parseErr := url.Parse(identityEndpoint)
+	if parseErr != nil {
+		return nil, fmt.Errorf("couldn't parse %q: %s", identityEndpoint, parseErr)
+	}
+
+	msiParameters := msiEndpoint.Query()
+	msiParameters.Set(apiVersionQueryParameterName, azureArcAPIVersion)
+	resource = strings.TrimSuffix(resource, "/.default")
+	msiParameters.Set(resourceQueryParameterName, resource)
+
+	if len(claims) > 0 {
+		msiParameters.Set("claims", claims)
+	}
+
+	msiEndpoint.RawQuery = msiParameters.Encode()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, msiEndpoint.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating http request %s", err)
+	}
+	req.Header.Set(metaHTTPHeaderName, "true")
+	return req, nil
+}
+
+// GetEnvironmentVariables returns the identity and IMDS endpoints
+func getAzureArcEnvironmentVariables() (string, string) {
```

Comment:
nit: name of the function is misleading. This tries to detect the azure arc env. Something like detectAzureArcEnv or validateAzureArcEnv maybe?

Reply:
Hmm, well the function is doing that but it is also returning the identity and imds vars, maybe validateAndRetrieveAzureArcEnvVars?

Reply:
Based on another comment I merged the 2 functions and renamed it to getAndValidateAzureArcEnvVars